core/src/main/java/org/elasticsearch/action/update/UpdateRequest.java

```diff
@@ -893,4 +894,31 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public String toString() {
+        String res = "update {[" + index + "][" + type + "][" + id + "]";
```

Comment:
There's lots of appending going on - could you use a `StringBuilder` here?

Reply:
Fixed. As far as I see, the readability is affected, and as far as I know, java compiler will replace it with StringBuilder automatically now.